pkg/resource/repository/custom_update_api.go

```diff
@@ -193,3 +199,55 @@ func (rm *resourceManager) deleteLifecyclePolicy(
 	}
 	return desired, nil
 }
+
+// syncRepositoryTags updates an ECR repository tags.
+func (rm *resourceManager) syncRepositoryTags(
+	ctx context.Context,
+	latest *resource,
+	desired *resource,
+) (err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.syncRepositoryTags")
+	defer exit(err)
+
+	added, updated, removed := computeTagsDelta(latest.ko.Spec.Tags, desired.ko.Spec.Tags)
+
+	// There are no API calls to update an existing tag. To update a tag we will have to first
+	// delete it and then recreate it with the new value.
```

Comment:
If you call `TagResource` with the key of an existing tag, will that not update it? I see that behaviour is not documented in the [docs](https://docs.aws.amazon.com/sdk-for-go/api/service/ecr/#ECR.TagResource)

Reply:
Good call! I just tested `TagResource` and apparently it updates the tags aswell! :+1: